packages/kitsu-core/src/deserialise/index.js

```diff
@@ -11,7 +11,7 @@ import { linkRelationships } from '../linkRelationships'
 function deserialiseArray (array) {
   for (let value of array.data) {
     if (array.included) value = linkRelationships(value, array.included)
-    if (value.relationships) value = linkRelationships(value)
+    value = linkRelationships(value, array.data)
```

Comment:
i initially wrote it like this, but due to `array.data` always being present, the `if` would always run.
```
if (array.data) value = linkRelationships(value, array.data)
if (value.relationships) value = linkRelationships(value)
```
due to `linkRelationships` removing the `relationships` key, the following `if (value.relationships)` would never run.

Reply:
nevermind this. i just realized that this approach breaks when `included` exists.
i've amended the solution and spec to handle this.